### DISCUSSION TITLE
canonical url

### DISCUSSION THREAD

**Discussion by Arhell** [statiqdev/Discussions#13] [Q&A] [answered]
And it is possible to display canonical url in head?
**example:**

- main page` https://test.com` `<link rel="canonical" href="https://test.com">`
- second https:`//test.com/second` `<link rel="canonical" href="https://test.com/second">`
etc

More example for jekyll short code `{{ site.url }}{{ page.url }}`

**Comment by daveaglick** [comment, marked as answer]
Sure! If you call `IDocument.GetLink(true)` (or `IExecutionContext.GetLink(document, true)`) it'll produce a link to the current document that includes [the `Host` setting](https://statiq.dev/framework/configuration/settings).

For example, to add the canonical link to a Razor page or layout you'd write:

```
<link rel="canonical" href="@Document.GetLink(true)">
```

It's up to the theme or layout to add this and I've added it to the [CleanBlog theme](https://github.com/statiqdev/CleanBlog).

**Reply by Arhell** [reply]
does not come out, writes
[ERRO] Content/PostProcess > RenderPostProcessTemplates > ExecuteIf > Error while executing module MirrorResources in Content/PostProcess: Response status code does not indicate success: 404 (Not Found).

**Reply by daveaglick** [reply]
How interesting! I mean, crashing probably isn't ideal 🤣 but it's still a little crazy to me the ways in which parts of the system interact sometimes. I suspect what's going on is that the `MirrorResources` module sees the `<link>` and is trying to treat it as a resource that can be mirrored since it's an absolute URL (`MirrorResources` is at least smart enough to avoid relative links). I'll need to add an exception for `rel="canonical"` since those should never be mirrored. I'll get it in the next release, and I'll try to get that out the door quickly. In the mean time you can try adding a `data-no-mirror` attribute to the `<link>` and that should avoid mirroring.

**Reply by Arhell** [reply]
Can you show an example how I can implement this?
I want to do here https://github.com/dotnet-foundation/website

**Reply by daveaglick** [reply]
Sure thing - it’ll end up looking like:

```
<link rel="canonical" href="@Document.GetLink(true)" data-no-mirror>
```

**Reply by Arhell** [reply]
Thank you so much!
